File: common/src/main/java/com.abc/kafka/Sender.java
package com.abc.kafka;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.kafka.support.SendResult;
import org.springframework.stereotype.Component;
import org.springframework.util.concurrent.ListenableFuture;
import org.springframework.util.concurrent.ListenableFutureCallback;

@Component
public class Sender {

    @Autowired
    private KafkaTemplate<String, String> kafkaTemplate;

    private final Logger log = LoggerFactory.getLogger(this.getClass());

    public void sendMessage(String topic, String message) {
        ListenableFuture<SendResult<String, String>> future = kafkaTemplate.send(topic, message);
        future.addCallback(
                new ListenableFutureCallback<SendResult<String, String>>() {
                    @Override
                    public void onFailure(Throwable ex) {
                        log.warn("Fail to send request", ex);
                    }

                    @Override
                    public void onSuccess(SendResult<String, String> result) {
                        log.info("Send message='{}' with offset={}", message, result.getRecordMetadata().offset());
                    }
                });
    }

}

File: fraud_decision_agent/src/main/java/com.abc/kafka/Sender.kt
package com.abc.kafka

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.kafka.support.SendResult
import org.springframework.stereotype.Component
import org.springframework.util.concurrent.ListenableFutureCallback

@Component
class Sender {

    @Autowired
    lateinit var kafkaTemplate: KafkaTemplate<String, String>

    fun sendMessage(topic: String, message: String) {
        val future = kafkaTemplate.send(topic, message)

        // you can register a callback with the listener to receive the result
        // of the send asynchronously
        future.addCallback(
                object : ListenableFutureCallback<SendResult<String, String>> {
                    override fun onSuccess(
                            result: SendResult<String, String>) {
                        println("sent message='$message' with offset=${result.getRecordMetadata().offset()}")
                    }
                    override fun onFailure(ex: Throwable) {
                        println("Failure")
                    }
                })
    }
}